x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/LifecyclePolicyUtils.java

```diff
@@ -35,32 +36,39 @@ public class LifecyclePolicyUtils {
     private LifecyclePolicyUtils() {};
 
     /**
-     * Loads a built-in index lifecycle policy and returns its source.
+     * Loads a built-in index lifecycle policy and returns its source with the given parser.
      */
     public static LifecyclePolicy loadPolicy(
         String name,
         String resource,
         Map<String, String> variables,
-        NamedXContentRegistry xContentRegistry
+        NamedXContentRegistry xContentRegistry,
+        XContentType contentType
     ) {
         try {
             String source = TemplateResources.load(resource);
             source = replaceVariables(source, variables);
             validate(source);
 
-            try (
-                XContentParser parser = XContentType.JSON.xContent()
-                    .createParser(XContentParserConfiguration.EMPTY.withRegistry(xContentRegistry), source)
-            ) {
-                LifecyclePolicy policy = LifecyclePolicy.parse(parser, name);
-                policy.validate();
-                return policy;
-            }
+            return parsePolicy(source, name, xContentRegistry, contentType);
         } catch (Exception e) {
             throw new IllegalArgumentException("unable to load policy [" + name + "] from [" + resource + "]", e);
         }
     }
 
+    public static LifecyclePolicy parsePolicy(
+        String rawPolicy,
+        String name,
+        NamedXContentRegistry xContentRegistry,
+        XContentType contentType
+    ) throws IOException {
+        XContentParser parser = contentType.xContent()
+            .createParser(XContentParserConfiguration.EMPTY.withRegistry(xContentRegistry), rawPolicy);
```

Comment:
This parser was originally within a try-with-resources loop, which ensures it's closed after being used. Can you please re-add the try-with-resources block?